apm-sniffer/apm-sdk-plugin/feign-default-http-9.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/feign/http/v9/PathVarInterceptor.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.apm.plugin.feign.http.v9;
+
+import feign.RequestTemplate;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.EnhancedInstance;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.InstanceMethodsAroundInterceptor;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.MethodInterceptResult;
+
+import java.lang.reflect.Method;
+
+/**
+ * {@link PathVarInterceptor} intercept the Feign RequestTemplate args resolve ;
+ *
+ * @author qiyang
+ */
+public class PathVarInterceptor implements InstanceMethodsAroundInterceptor {
+
+    static final ThreadLocal<String> URL_CONTEXT = new ThreadLocal<String>();
+
+    /**
+     * Get the {@link RequestTemplate#url()} before feign.ReflectiveFeign.BuildTemplateByResolvingArgs#resolve(Object[], RequestTemplate, Map)
+     *  put it into the {@link PathVarInterceptor#URL_CONTEXT}
+     *
+     * @param method intercept method
+     * @param result change this result, if you want to truncate the method.
+     */
+    @Override public void beforeMethod(EnhancedInstance objInst, Method method, Object[] allArguments,
```

Comment:
`@Override` The format suggested here is adjusted

Reply:
This is optional.